clippy_lints/src/empty_with_brackets.rs

```diff
@@ -142,6 +199,20 @@ fn has_no_fields(cx: &EarlyContext<'_>, var_data: &VariantData, braces_span: Spa
     has_no_ident_token(braces_span_str.as_ref())
 }
 
+fn check_expr_for_enum_as_function(expr: &Expr<'_>) -> Option<DefId> {
```

Comment:
This should check that the `DefId` is local and return `LocalDefId`. We are only interested in local variants so we don't need to insert `Used` variants from other crates into the map.